opennms-dao/src/main/java/org/opennms/netmgt/dao/support/AbstractTopLevelResourceType.java

```diff
@@ -47,7 +48,10 @@ public boolean isResourceTypeOnParent(OnmsResource parent) {
 
     @Override
     public List<OnmsResource> getResourcesForParent(OnmsResource parent) {
-        return getTopLevelResources();
+        if (parent == null) {
```

Comment:
Then, why having that parameter instead of modifying the method signature to be the following ?

 public List<OnmsResource> getResourcesForParent();